src/metabase/analytics/stats.clj

```diff
@@ -362,37 +348,64 @@
 
 ;;; Execution Metrics
 
-(defn- summarize-executions
-  "Summarize `executions`, by incrementing approriate counts in a summary map."
-  ([]
-   (summarize-executions (t2/reducible-query {:select [:executor_id :running_time [[:case
-                                                                                    [:= :error nil ] false
-                                                                                    [:= :error "" ] false
-                                                                                    :else true] :has_error]]
-                                              :from   [:query_execution]})))
-  ([executions]
-   (reduce summarize-executions {:executions 0, :by_status {}, :num_per_user {}, :num_by_latency {}} executions))
-  ([summary execution]
-   (-> summary
-       (update :executions u/safe-inc)
-       ;; MYSQL is returning true as 1, false as 0! what!
-       (update-in [:by_status (if (#{1 true} (:has_error execution))
-                                "failed"
-                                "completed")] u/safe-inc)
-       (update-in [:num_per_user (:executor_id execution)] u/safe-inc)
-       (update-in [:num_by_latency (bin-large-number (/ (:running_time execution) 1000))] u/safe-inc))))
-
-(defn- summarize-executions-per-user
-  "Convert a map of `user-id->num-executions` to the histogram output format we expect."
-  [user-id->num-executions]
-  (frequencies (map bin-large-number (vals user-id->num-executions))))
+(defn- execution-metrics-sql []
+  (let [thirty-days-ago (str "CAST('" (t/minus (t/offset-date-time) (t/days 30)) "' AS TIMESTAMP)")]
+    (str/join
+     "\n"
+     ["WITH user_executions AS ("
```

Comment:
Maybe it would be nicer just to put this big ol query in its own `.sql` file (maybe something like `src/metabase/analytics/stats/execution-metrics.sql`) and then `slurp` the `io/resource` so you don't have to `str/join` this big hairy query every time we run it. For thirty-days ago couldn't you just include that in the query itself? Something like

```sql
current_timestamp - interval '30 days'
```

?

Or just pass it as a JDBC parameter like `CAST(? AS timestamp with time zone)`

Reply:
I would agree but it turns out we need database-specific strings to implement this properly, which means we would need three `.sql` files. See 
 the other comment for why: https://github.com/metabase/metabase/pull/46586#discussion_r1721697566. We need to use SQL strings or HoneySQL. I'm leaning towards preferring SQL strings over HoneySQL for this query simply because there's not much string manipulation here so HoneySQL won't add much value.